Add support for Next.js >= 13.2

yoieh opened the pull request:
Fixes #4 

`ensureApiPage` was removed in `v13.1.7-canary.14`

https://github.com/vercel/next.js/pull/45716/files#diff-2f6fd5a9538abfaf3d5b76a82a8d1ffa029f84a36f0cdae5460520c5cfc2db6bL1402

yoieh committed:
Addes support for NetxtJS > 13.1.6

fixed: #4 

`ensureApiPage` was removed in v13.1.7-cannary.14

https://github.com/vercel/next.js/pull/45716/files#diff-2f6fd5a9538abfaf3d5b76a82a8d1ffa029f84a36f0cdae5460520c5cfc2db6bL1402

brhx reviewed (changes requested):
Excellent spot, thank you! Please could you also update the README.md with the tested version?

yoieh committed:
Update README.md with tested version

yoieh commented:
I have updated the `Compatibility` section of the readme.
This still doesn't give full support for the new app directory with the new [Route Handlers](https://beta.nextjs.org/docs/routing/route-handlers).

brhx committed:
Run prettier format

brhx committed:
Import typedefs for both versions of next dev server

sam3d commented:
The typescript type check fails because it needs to be aware of both possible versions of `NextDevServer`. I'd like to ensure it doesn't

yoieh commented:
I see,
There are more issues here. 

`ensurePage` only exists on the `DevServer` in version 13.3.0 and not at all `NextNodeServer`.
This PR will never work in production.

yoieh committed:
Revert "Import typedefs for both versions of next dev server"

This reverts commit f329a53182558c316bc14d578c977ff2f42a6f44.

yoieh committed:
Changed out ensurePage and to just use ensureEdgeFunction

Changed ensurePage to use ensureEdgeFunction

Change it to just ensureEdgeFunction

yoieh commented:
I still kept the `ensureApiPage` for now.. 
but it look like you may be able to just use the `ensureEdgeFunction` from here on.

Not sure when i was added so can not garanti it works with a version lower than 13.1.6.
But this will "make typescript not fail".

brhx reviewed (changes requested):
Oooh I like this solution! Please could you change it to just `ensureEdgeFunction` and I will give it a test and merge?
